File: Day 4/240. Search a 2D Matrix II.py
class Solution(object):
    def searchMatrix(self, matrix, target):
        r = len(matrix)
        c = len(matrix[0])
        a = 0
        b = c - 1
        while a < r and b >= 0:
            if target == matrix[a][b]:
                return True
            elif target > matrix[a][b]:
                a = a + 1
            else:
                b = b - 1
        return False

File: Day 004/240. Search a 2D Matrix II.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int r = matrix.length;
        int c = matrix[0].length;
        int a = 0;
        int b = c - 1;
        while (a < r && b >= 0) {
            if (target == matrix[a][b]) {
                return true;
            } else if (target > matrix[a][b]) {
                a++;
            } else {
                b--;
            }
        }
        return false;
    }
}
